src/dbnode/persist/fs/retriever_test.go

```diff
@@ -674,6 +676,87 @@ func TestBlockRetrieverOnlyCreatesTagItersIfTagsExists(t *testing.T) {
 	require.NoError(t, err)
 }
 
+// TestBlockRetrieverDoesNotInvokeOnRetrieveWithGlobalFlag verifies that the block retriever
+// does not invoke the OnRetrieve block if the global CacheBlocksOnRetrieve is not enabled.
+func TestBlockRetrieverDoesNotInvokeOnRetrieveWithGlobalFlag(t *testing.T) {
+	testBlockRetrieverDoesNotInvokeOnRetrieve(t, false, true)
+}
+
+// TestBlockRetrieverDoesNotInvokeOnRetrieveWithNamespacesFlag verifies that the block retriever
+// does not invoke the OnRetrieve block if the namespace-specific CacheBlocksOnRetrieve is not enabled.
+func TestBlockRetrieverDoesNotInvokeOnRetrieveWithNamespaceFlag(t *testing.T) {
+	testBlockRetrieverDoesNotInvokeOnRetrieve(t, true, false)
```

Comment:
Probably worth doing `testBlockRetrieverDoesNotInvokeOnRetrieve(t, false, false)` as well for full coverage of all permutations

Reply:
Any maybe the opposite? I.e. caching flags both true do invoke OnRetrieve?

Reply:
Yeah, figured that's tested implicitly by existing tests that checked cache writes, but may be worth writing up an explicit test as well for completeness

Reply:
Yeah, @arnikola 's got the right read of it. But should probably have an explicit test just in case. Will add.